app/controllers/admin/users_controller.rb

```diff
@@ -47,11 +47,7 @@ def user_params
   end
 
   def get_users
-    @users = User.by_user_name(params[:name])
-                 .by_status(params[:status])
-                 .by_role(params[:role])
-                 .order_by_post_count(params[:order_by])
-                 .order_created_at
-                 .page(params[:page]).per Settings.user.page
+    @q = User.ransack params[:q]
```

Comment:
để tên khác params[:q] dc k e?

Reply:
dạ tham số cố định để search của ransack là q ạ và nếu thay đổi tên khác như param[:search] thì sẽ trả về nil ạ

Reply:
vậy làm sao để a đổi được, ko dùng q nữa

Reply:
dạ thì anh có thể config trong file config/initializers/ransack.rb:
Ransack.configure do |config|

  # Change default search parameter key name.
  # Default key name is :q
  config.search_key = :query

  # Raise errors if a query contains an unknown predicate or attribute.
  # Default is true (do not raise error on unknown conditions).
  config.ignore_unknown_conditions = false

  # Globally display sort links without the order indicator arrow.
  # Default is false (sort order indicators are displayed).
  # This can also be configured individually in each sort link (see the README).
  config.hide_sort_order_indicators = true

end

=> config.search_key = :query thì ta có thể config lại tham số q ạ